src/improvements/doc/TEMPLATE_ARCHITECTURE.md

```diff
@@ -0,0 +1,156 @@
+
+### 2. Configuration File Architecture
+
+Each task requires a `config.toml` file with two main components:
+
+1. L2 Chain Configuration
+2. Template Selection
+
+#### L2 Chain Configuration
+
+Every config file must specify the L2 chains that the task will interact with:
+
+```toml
+l2chains = [
+    {name = "Orderly", chainId = 291},
+    {name = "Metal", chainId = 1750}
+]
+```
+
+#### Template Selection
+
+Templates are specified using the `templateName` parameter:
+
+```toml
+templateName = "GasConfigTemplate"  # Uses src/improvements/template/GasConfigTemplate.sol
+```
+
+### 3. Task Status Definitions
+
+Tasks can have the following statuses in their README files:
+
+- `DRAFT, NOT READY TO SIGN`: Initial development stage, not ready to sign
+- `CONTINGENCY TASK, SIGN AS NEEDED`: Task that may be signed based on specific conditions, such as emergency rollback.
+- `READY TO SIGN`: Task ready for signature
+- `SIGNED`: Task signed and ready for execution
+- `EXECUTED`: Task completed successfully
+- `CANCELLED`: Execution failed or abandoned
+
+### 4. Template-Specific Configuration
+
+Each template type has its own configuration structure. For example:
+
+```toml
+# Gas Config Template
+[gasConfigs]
+gasLimits = [
+    {chainId = 291, gasLimit = 100000000},
+    {chainId = 1750, gasLimit = 100000000}
+]
+
+# Dispute Game Template
+implementations = [{
+    gameType = 0,
+    implementation = "0xf691F8A6d908B58C534B624cF16495b491E633BA",
+    l2ChainId = 10
+}]
+```
+
+## Validation System
+
+The template system includes built-in validation:
+
+1. Configuration Validation
+   - Verifies TOML syntax
+   - Checks required fields
+   - Validates chain IDs
+   - Ensures parameter types match expectations
+
+2. State Change Validation
+   - Verifies changes match expected values
+
+3. Security Checks
+   - Verifies permissions
+   - Validates address formats
+   - Checks value ranges
+   - Check that only the expected storage slots are modified
+
+## Network Organization
+
+Tasks are organized by network in the tasks directory:
+
+1. Ethereum Mainnet (`eth/`)
+   - Production-ready tasks
+   - Strict validation requirements
+   - Full security review needed
+
+2. Sepolia Testnet (`sep/`)
+   - Testing and validation
+   - Development and staging
+   - Less strict security requirements
+
+3. Optimism Networks
+   - `oeth/`: Optimism Ethereum tasks
+   - `opsep/`: Optimism Sepolia tasks
+   - Network-specific configurations
+
+Each network directory follows a numbered sequence for tasks (e.g., 001, 002) to maintain chronological order and dependencies.
+
+## Best Practices
+
+1. Configuration Organization
+   - Keep network-specific parameters separate
+   - Use clear parameter names
+   - Document all configuration options
+
+2. Validation Rules
+   - Assert all expected state changes occurred correctly
+   - Assert no unexpected state changes occurred
+
+## Common Fields Reference
```

Comment:
Lets remove this, we've mentioned this a lot in various markdown files at this point.